catroid/src/main/java/org/catrobat/catroid/utils/FormatNumberUtil.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.utils;
+
+import java.math.BigDecimal;
+
+public final class FormatNumberUtil {
+
+	// Suppress default constructor for noninstantiability
+	private FormatNumberUtil() {
+		throw new AssertionError();
+	}
+
+	public static String cutTrailingZeros(String number) {
+
+		BigDecimal decimal = new BigDecimal(number);
+		decimal = decimal.stripTrailingZeros();
+
+		if (decimal.compareTo(BigDecimal.ZERO) == 0) {
+			decimal = BigDecimal.ZERO;
```

Comment:
Please readd comment that this is necessary because of a bug in the BigDecimal implementation

Reply:
Or is there a test that catches this?

Reply:
ok, then leave the comment here, but remove the one in line 30 :-)

Reply:
done